src/apps/AraConsole/AraConsole.js

```diff
@@ -0,0 +1,293 @@
+      log('permParams', permParams, initParams)
+
+      const web3 = new Web3(getInjectedProvider())
+
+      if (!web3) {
+        toast('You need to have a Dapp browser extension for this command')
+      }
+
+      const kernelProxyAddress = apps.find(
+        app => app.name.toLowerCase() === 'kernel'
+      ).proxyAddress
+
+      const appId = apps.find(app => app.name.toLowerCase() === appName).appId
+
+      const ensDomain = await wrapper.ens.resolve(appId)
+      log('ensDomain', ensDomain)
+
+      const repoContent = await wrapper.apm.fetchLatestRepoContent(ensDomain, {
+        fetchTimeout: 3000,
+      })
+      const { abi, contractAddress, roles } = repoContent
+
+      const { name, type, inputs } = abi.find(
+        ({ name }) => name === 'initialize'
+      )
+
+      const encodedInitializeFunc = web3.eth.abi.encodeFunctionCall(
+        {
+          name,
+          type,
+          inputs,
+        },
+        [...initParams]
+      )
+
+      const path = await wrapper.getTransactionPath(
+        kernelProxyAddress,
+        'newAppInstance(bytes32,address,bytes,bool)',
+        [appId, contractAddress, encodedInitializeFunc, false]
+      )
+
+      // Get the second to last item in the path, as it is the account that will execute kernel.newAppInstance
+      const scriptExecutor = path[path.length - 2].to
+      const counterfactualAppAddr = await wrapper.kernelProxy.call(
+        'newAppInstance',
+        appId,
+        contractAddress,
+        encodedInitializeFunc,
+        false,
+        { from: scriptExecutor }
+      )
+
+      const installAppIntent = [
+        [
+          kernelProxyAddress,
+          'newAppInstance(bytes32,address,bytes,bool)',
+          [appId, contractAddress, encodedInitializeFunc, false],
+        ],
+      ]
+
+      const aclProxyAddress = apps.find(app => app.name.toLowerCase() === 'acl')
+        .proxyAddress
+
+      const permissionIntents = permParams.map(([role, from, to]) => {
+        const roleBytes = roles.find(availableRole => availableRole.id === role)
+          .bytes
+
+        return [
+          aclProxyAddress,
+          'createPermission',
+          [to, counterfactualAppAddr, roleBytes, from],
+        ]
+      })
+
+      const intentBasket = [...installAppIntent, ...permissionIntents]
+      log(intentBasket, permissionIntents)
+      const {
+        path: pathForBasket,
+        transactions,
+      } = await wrapper.getTransactionPathForIntentBasket(intentBasket)
+      log('DONE', pathForBasket, transactions)
+      if (Array.isArray(pathForBasket) && pathForBasket.length) {
+        const transId = await wrapper.performTransactionPath(pathForBasket)
+        log(transId)
+      } else if (Array.isArray(transactions) && transactions.length) {
+        for (const transaction of transactions) {
+          await wrapper.performTransactionPath([transaction])
+        }
+      } else {
+        await wrapper.performTransactionPath([])
+      }
```

Comment:
Should we move this all into a `async performIntents()` function, so we can reuse it in `handleDaoExec()`?

Reply:
Sounds good to me; cc @sohkai